ui/src/app/pages/data/data-set/genomic-extraction-modal.tsx

```diff
@@ -25,7 +39,31 @@ interface Props {
 export const GenomicExtractionModal = ({
     dataSet, workspaceNamespace, workspaceFirecloudName, closeFunction, title, cancelText, confirmText}: Props) => {
   const [error, setError] = useState(false);
-  const [loading, setLoading] = useState(false);
+  const [loading, setLoading] = useState(true);
+
+  const genomicExtractions = useStore(genomicExtractionStore);
+  const extractsForWorkspace = genomicExtractions && genomicExtractions[workspaceNamespace] || [];
+  const extractsForDataset = fp.filter((extract: GenomicExtractionJob) => extract.datasetName === dataSet.name, extractsForWorkspace);
+
+  useEffect(() => {
+    if (!(workspaceNamespace in genomicExtractions)) {
+      dataSetApi().getGenomicExtractionJobs(workspaceNamespace, workspaceFirecloudName)
+        .then(resp => updateGenomicExtractionStore(workspaceNamespace, resp.jobs))
+        .finally(() => setLoading(false));
+    } else {
+      setLoading(false);
+    }
+  }, [workspaceNamespace]);
+
+  const runningExtract = fp.flow(
+    fp.orderBy((extract: GenomicExtractionJob) => extract.submissionDate, 'desc'),
+    fp.find((extract: GenomicExtractionJob) => extract.status === TerraJobStatus.RUNNING)
+  )(extractsForDataset);
+  const succeededExtract = fp.flow(
+    fp.orderBy((extract: GenomicExtractionJob) => extract.completionTime, 'desc'),
+    fp.find((extract: GenomicExtractionJob) => extract.status === TerraJobStatus.SUCCEEDED)
+  )(extractsForDataset);
```

Comment:
UX question. if the user launched a job after a successful one and it failed, should we still show them in warning?

for example:
10 days ago - success job 
5 days ago - job failed
now - user tries to launch a job

I think both can make sense but I wanted to get your opinion.

Reply:
This is a question for Lou, I think - I'll ask